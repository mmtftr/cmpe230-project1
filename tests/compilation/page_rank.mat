# simple pageranking
# algorithm
matrix   A[3,3]
matrix   T[1,1]
vector   x[3]
vector   y[3]
scalar   r
scalar   i
A={0.5 0 0.5 0 0 0.5 0.5 1 0} 
x={1 1 1 }
for(i in 1:10:1) {
  y = A*x
  T = tr(y-x)*(y-x)
  r = sqrt(T[1,1]) 
  print(r)
  x=y
}
printsep()
print(x)