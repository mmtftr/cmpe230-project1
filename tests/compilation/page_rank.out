0.707107
0.612372
0.467707
0.385276
0.309359
0.250975
0.202824
0.164156
0.132784
0.107431
------------
1.214844
0.624023
1.161133
